pkg/utils/utils.go

```diff
@@ -627,12 +647,90 @@ func GetClusterFlavor(config *rest.Config) (constants.ClusterFlavor, error) {
 		}
 	}
 
-	clientset, err := kubernetes.NewForConfig(config)
+	clientset, err := GetKubeClientSet(config)
 	if err != nil {
 		return constants.Unknown, err
 	}
 
-	return GetCSIClusterType(clientset)
+	veleroNs, exist := GetVeleroNamespace()
+	if !exist {
+		fmt.Printf("The VELERO_NAMESPACE environment variable is empty, assuming velero as namespace\n")
+		veleroNs = constants.DefaultVeleroNamespace
+	}
+
+	return retrieveClusterFlavor(clientset, veleroNs)
+}
+
+func retrieveClusterFlavor(clientset kubernetes.Interface, veleroNs string) (constants.ClusterFlavor, error) {
```

Comment:
I believe the same thing can be applied to merge `retrieveClusterFlavor` and `RetrieveClusterFlavor` and change the its test a little bit as you did to `rerieveVcConfigSecret` and `RerieveVcConfigSecret`.

Reply:
Mentioned this in previous comments:
Just FYI, i've fixed this for RetrieveVcConfigSecret, however, retrieveClusterFlavor is a little different. Here, the method RetrieveClusterFlavor is called with veleroNs passed on, the velero ns is passed because for some reason during the backup-driver & data-manager startup the environment variable VELERO_NAMESPACE is not found, I suspect this has to do something with how velero adds the plugin. RetrieveClusterFlavor is exclusively called from install.go of backup-driver and data-manager, while the rest of the code in plugin determines the cluster flavor using GetClusterFlavor. Both these methods eventually call retrieveClusterFlavor for which I've written unit tests.

Reply:
Make sense. I didn't notice `retrieveClusterFlavor` is also consumed by other util funcs. Then no need to make the same merge of `retrieveClusterFlavor` and `RetrieveClusterFlavor`